src/main.c

```diff
@@ -444,7 +447,19 @@ void load_config(void)
 	}
 
 	if (cf_auth_type == AUTH_HBA) {
-		struct HBA *hba = hba_load_rules(cf_auth_hba_file);
+		struct IDENT *ident;
+		struct HBA *hba;
+
+		ident = ident_load_map(cf_auth_ident_file);
+
+		if (ident) {
+			if (parsed_ident)
```

Comment:
nit: this check is unnecessary, you can always free even if it is NULL (then free is a no-op)